compiler-core/src/erlang/tests/records.rs

```diff
@@ -93,6 +93,34 @@ pub fn get_name(person: Person) { person.name }
     );
 }
 
+#[test]
+fn record_accessor_multiple_variants() {
+    // We can access fields on custom types with multiple variants
+    assert_erl!(
+        "
+pub type Person {
+    Teacher(name: String, title: String)
+    Student(name: String, age: Int)
+}
+pub fn get_name(person: Person) { person.name }"
+    );
+}
+
+#[test]
+fn record_accessor_multiple_variants_positions_other_than_first() {
+    // We can access fields on custom types with multiple variants
+    // In positions other than the 1st field
+    assert_erl!(
+        "
+pub type Person {
+    Teacher(name: String, age: Int, title: String)
+    Student(name: String, age: Int)
+}
+pub fn get_name(person: Person) { person.name }
+pub fn get_age(person: Person) { person.age }"
+    );
```

Comment:
Could we also have a test like this where the first is different? Thanks

```gleam
pub type Person {
    Teacher(name: Nil, age: Int)
    Student(name: String, age: Int)
}
pub fn get_age(person: Person) { person.age }
```

Reply:
For both erlang and js?